content.js

```diff
@@ -168,20 +130,81 @@ async function sleep(ms){await new Promise(r=>setTimeout(r,ms||100));};`;
       return index;
     }
 
-    function convertActionToCode(action, recordIndex) {
+    function convertActionToCode(action, element, recordIndex) {
       let type = action.selector?.match(/\[type="(.+)"\]/) || "";
-      if (type) type = type[0];
+      if (type) {
+        type = type[0];
+      } else {
+        type = element.type;
+      }
       const setValue = dotValueForType(type) || "value";
       const selector = `${action.selector}${
-        action.index ? "[" + (action.index + 1) + "]" : ""
+        action.index ? ":nth-of-type(" + (action.index + 1) + ")" : ""
       }`;
-      const value = action.value.replace(/`/g, "\\`");
+      const value =
+        typeof action.value === "string"
+          ? action.value.replace(/`/g, "\\`")
+          : action.value;
```

Comment:
needed this to handle boolean `true` or `false` since you can't run `.replace()` on booleans, only on strings